File: Decorator/Golang/MonochromeDecorator.go
package main

import "fmt"

type MonochromeDecorator struct {
	PrinterDecorator
}

func (monochromeDecorator MonochromeDecorator) print()  {
	monochromeDecorator.PrinterDecorator.print()
	fmt.Println("Printed with monochrome color")
}

func NewMonochromeDecorator(printer IPrinter) MonochromeDecorator {
	return MonochromeDecorator{PrinterDecorator{printer}}
}

File: Decorator/PHP/MonochromeDecorator.php
<?php

require_once "PrinterDecorator.php";
require_once "IPrinter.php";

class MonochromeDecorator extends PrinterDecorator {
    public function __construct(IPrinter $printer)
    {
        parent::__construct($printer);
    }

    public function print() {
        parent::print();
        echo "Printed with monochrome color\n";
    }
}